File: app/src/main/java/indi/yume/app/passwordbox/renderer/ItemRenderer.java
package indi.yume.app.passwordbox.renderer;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.Bind;
import butterknife.ButterKnife;
import indi.yume.app.passwordbox.R;
import indi.yume.app.passwordbox.model.database.DataTable;
import indi.yume.app.passwordbox.view.CircleView;
import indi.yume.tools.adapter_renderer.BaseRenderer;

/**
 * Created by bush2 on 2016/4/4.
 */
public class ItemRenderer extends BaseRenderer<DataTable> {
    @Bind(R.id.title_text_view)
    TextView titleTextView;
    @Bind(R.id.website_text_view)
    TextView websiteTextView;
    @Bind(R.id.head_text_view)
    CircleView headTextView;

    @Override
    public void render() {
        DataTable dataTable = getContent();

        headTextView.setText(dataTable.getTitle());
        headTextView.setColor(dataTable.getLabelColor());
        titleTextView.setText(dataTable.getTitle());
        websiteTextView.setText(dataTable.getWebsite());
    }

    @Override
    protected View inflate(LayoutInflater layoutInflater, ViewGroup viewGroup) {
        return layoutInflater.inflate(R.layout.item_layout, viewGroup, false);
    }

    @Override
    protected void findView(View view) {
        ButterKnife.bind(this, view);
    }

    @Override
    protected void setListener(View view) {

    }
}

File: appkotlin/src/main/kotlin/indi/yume/view/appkotlin/renderer/ItemRenderer.kt
package indi.yume.view.appkotlin.renderer

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import indi.yume.tools.adapter_renderer.BaseRenderer
import indi.yume.view.appkotlin.R
import indi.yume.view.appkotlin.database.DataTable
import indi.yume.view.appkotlin.util.bindView
import indi.yume.view.appkotlin.view.CircleView
import org.jetbrains.anko.find
import kotlin.properties.Delegates

/**
 * Created by bush2 on 2016/4/4.
 */
class ItemRenderer : BaseRenderer<DataTable>() {
    var mainView: View by Delegates.notNull()

    val titleTextView: TextView   by lazy { mainView.find<TextView>(R.id.head_text_view) }
    val websiteTextView: TextView by lazy { mainView.find<TextView>(R.id.website_text_view) }
    val headTextView: CircleView  by lazy { mainView.find<CircleView>(R.id.head_text_view) }

    override fun render() {
        val dataTable = content

        headTextView.setText(dataTable.title)
        headTextView.setColor(dataTable.labelColor)
        titleTextView.setText(dataTable.title)
        websiteTextView.setText(dataTable.website)
    }

    override fun inflate(layoutInflater: LayoutInflater, viewGroup: ViewGroup): View {
        return layoutInflater.inflate(R.layout.item_layout, viewGroup, false)
    }

    override fun findView(view: View) {
        mainView = view
    }

    override fun setListener(view: View) {

    }
}
